rosetta-source/src/main/rosetta/observable-asset-enum.rosetta

```diff
@@ -29,6 +29,13 @@ enum InformationProviderEnum: <"The enumerated values to specify the list of inf
 enum PriceExpressionEnum: <"Enumerated values to specify whether the price is expressed in absolute or relative terms.">
     AbsoluteTerms <"The price is expressed as an absolute amount.">
     PercentageOfNotional <"The price is expressed in percentage of the notional amount.">
+    PercentageOfInitialSpotPrice <"The price is expressed in percentage of the Initial Valuation Price.">
+    PercentageOfInitialValuationPrice <"The price is expressed in percentage of the Initial Valuation Price.">
+    PercentageOfInterimValuationPrice <"The price is expressed in percentage of the Interim Valuation Price.">
+    PercentageOfFinalValuationPrice <"The price is expressed in percentage of the Final Valuation Price.">
+    PercentageOfFirstObservationPeriodPrice <"The price is expressed in percentage of the First Observation Period Price.">
+    PercentageOfObservationPeriodPrice<"The price is expressed in percentage of each Observation Period Price.">
+    PercentageOfLastObservationPeriodPrice <"The price is expressed in percentage of the Last Observation Period Price.">
```

Comment:
do not agree, for two main reasons : 
1 - if this is not the purpose of current type PriceExpression, please explain what is the meaning of "PercentageOfNotional", if not to represent to what the "percentage" is is relative to i.e. "the notional", and therefore this logic shall naturally be extended to other kinds...
2 - that is a deep business requirement, you cannot avoid having such information in CDM : mostly all termsheets are for good reaons defining prices wth various "price expression", notably and mostly as percentage of other prices which as per market practice, are quite limited to the more "usual" ones e.g; "valuation prices" or "spot prices" taken as reference... so say PriceExpression is not the right place, then what is your suggestion for modelling all such attributes (which are defintly attribute of a Price per se..)
3 - an expression that is "Percentage in all cases" does not mean anything per se : "percentage **of what**" ? this makes all the difference for the business : a strike being "percentage" with no further specifications ... what does it mean , that is critical to be explicit what a strike is a percentage of...

Reply:
your proposal will result in agnostic kind of price "multiplied" by another agnostic kind of price, that is not what we want because we want to refer to kind of price which resonates with business and mostly used applicable documentation in ISDA Def or other kinds, whatever is it eventually, fact is that SpotPrice, KnockPrice, etc; and all the other ones I'm proposing to have are very typical ones mostly used in trade agreement terms, notably (not only) for one being % of another one, etc. 
so defintly an Enum list makes senses whenever values at stake need to refer to well known "concepts" fixed by business... also in your feedback you do not answer to one of my question : why then having "PercentageOfNotional" but not to accept having "PercentageOf[KindOfPrice"] ? that is not consistent... if puspoe of PriceExpressionENum in only to differentiate absolute and percentage, then could be a boolean, not a list, and still problem statement would remain... a "percentage" of what ?

eventually, i could imagine having en Enum type being epxlicit about what we need here, say type Enum "PercentageRelativeToEnum", which values would be the one i have propsed - what do you think ?